clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -449,9 +450,9 @@ public void close() {
     @Test
     public void testCloseOldestConnection() throws Exception {
         String id = "0";
-        blockingConnect(id);
-
-        time.sleep(6000); // The max idle time is 5000ms
+        selector.connect(id, new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE);
+        selector.poll(0);
```

Comment:
Note that `connect()` is synchronous method here. It might not complete with the first poll event. That is the reason it is tested in a while loop inside `blockingConnect`.

My suggestion would be to keep using blockingConnect with a (new) timeout to terminate while loop in it similar to implementation of `NetworkTestUtils.waitForChannelReady`.

Reply:
Thanks @divijvaidya for looking at this. I need to fix this and `testCloseOldestConnectionWithMultiplePendingReceives`.